File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        // return the message
        //int find = line.find(":");
        return line.substr(line.find(":")+2);
    }
    
    std::string log_level(std::string line) {
        // return the log level
        return line.substr(line.find("[")+1, line.find("]")-1);
    }
    
    std::string reformat(std::string line) {
        // return the reformatted message
        return message(line) + " (" + log_level(line) + ")";
    }
}  // namespace log_line

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String[] splitStr = logLine.split(" ", 2);
        return splitStr[1].trim();
    }

    public static String logLevel(String logLine) {
        if (logLine.contains("ERROR")){
            return "error";
        }
        if (logLine.contains("WARNING")){
            return "warning";
        }
        return "info";
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}
